internal/controller/controller.go

```diff
@@ -115,6 +116,15 @@ func StartControllers(ctx context.Context, config *rest.Config, logger logr.Logg
 	if err = mgr.Start(ctx); err != nil { // This blocks until the manager is stopped.
 		return fmt.Errorf("failed to start controller manager: %w", err)
 	}
+
+	// Have a token exchange handler per provider type.
+	handler, err := oidc.NewOIDCTokenExchange(&logger, c, aigv1a1.BackendSecurityPolicyTypeAWSCredentials)
+	if err != nil {
+		return fmt.Errorf("failed to create OIDC handler: %w", err)
+	}
+
+	go handler.RefreshCredentials(ctx)
+
```

Comment:
these code are not called since mgr.Start is blocking.

Reply:
I think you use let the manager Add your runable
https://github.com/kubernetes-sigs/controller-runtime/blob/v0.20.1/pkg/manager/manager.go#L52-L60

Reply:
I think you can use Manager to Add a Runable 
https://github.com/kubernetes-sigs/controller-runtime/blob/v0.20.1/pkg/manager/manager.go#L56-L60

and you don't need 
```
go handler.RefreshCredentials(ctx)
```

Manager can filter by
```
//reconciliation is only triggered by a GenericEvent generated by a time ticker
		WithEventFilter(predicate.Funcs{
			GenericFunc: func(event.GenericEvent) bool { return true },
````
We can write a tick event source to generate tick time event. I can find our internal example for you